### DISCUSSION TITLE
Problem removing dead code in async function

### DISCUSSION THREAD

**Discussion by SimonLundberg** [parcel-bundler/parcel#5665] [Q&A]
I'm building a web app with multiple build targets depending on needed functionality, so I'm setting environment variables for each target and reference those inside my JS files, like so:

`if (process.env.<VARIABLE> == "true") {
//do stuff...
}`

It mostly works fine. If <VARIABLE> isn't "true", Parcel strips out the code inside the if statement completely. However, some of my functions use the async/await pattern. In these functions, no code is removed at all. When using the .then pattern, it works as expected.

Here is an example:

```javascript
async function bar() {
    if (process.env.BAR == "true") { // Assume BAR is not "true"
        let bar = new Promise((resolve, _) => {
            resolve("bar");
        });
        bar = await bar;
        console.log("this code IS NOT stripped from the built file");
    }
}

function baz() {
    if (process.env.BAZ == "true") { // Assume BAZ is not "true"
        let baz = new Promise((resolve, _) => {
            resolve("baz");
        });
        baz.then(() => {
            console.log("this code IS stripped from the built file");
        })
    }
}
```

Is this a known limitation? A bug? Is there a way to work around it?

Much obliged!

Edit: the plot thickens — the following code works as expected! It is basically the same as the original bar, except it wraps the log in an additional function. I'm sorry but this makes _zero_ sense to me. 

```javascript
async function bar() {
    if (process.env.BAR == "true") {
        let bar = new Promise((resolve, _) => {
            resolve("bar");
        });
        const process = (thing) => {
            console.log("by wrapping the console.log in this arrow function, we fix the problem???");
        };
        bar = await bar;
        process(bar);
    }
}
```

**Comment by mischnic** [comment]
They are removed for me (taking these functions, adding `console.log(bar, baz, foo);` to the file). Both when async/await is transpiled and when it's not
